### PR TITLE
JDK-8293493: Signal Handlers printout should show signal block state

### PR DESCRIPTION
**Pull request by tstuefe**
I needed this recently. We print out the signal block mask, which is fine, but more interesting would be the existing block state at the point of the crash/assertion.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8293493](https://bugs.openjdk.org/browse/JDK-8293493): Signal Handlers printout should show signal block state

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Johan Sjölen](https://openjdk.org/census#jsjolen) (@jdksjolen - Author)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10199/head:pull/10199` \
`$ git checkout pull/10199`

Update a local copy of the PR: \
`$ git checkout pull/10199` \
`$ git pull https://git.openjdk.org/jdk pull/10199/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10199`

View PR using the GUI difftool: \
`$ git pr show -t 10199`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10199.diff">https://git.openjdk.org/jdk/pull/10199.diff</a>

</details>

### CODE REVIEW

**Comment by tstuefe** [comment]
Friendly ping. Change is really simple.

File: src/hotspot/os/posix/signals_posix.cpp [hunk 1]
```diff
@@ -1414,6 +1413,11 @@ void PosixSignals::print_signal_handler(outputStream* st, int sig,
   sigaction(sig, NULL, &current_act);
 
   print_single_signal_handler(st, &current_act, buf, buflen);
+
+  sigset_t thread_sig_mask;
+  if (::pthread_sigmask(/* ignored */ SIG_BLOCK, NULL, &thread_sig_mask) == 0) {
```
Comment (dholmes-ora) [inline src/hotspot/os/posix/signals_posix.cpp:1418]: Nit: no need for `::` scoping (there is one other existing unnecessary use as well you could remove :) ).

**Review by dholmes-ora — APPROVED** [review]
This seems okay though may be of limited use if the crash is from a signal as IIRC that in itself will result in various signals getting blocked whilst we remain in the signal-handler context.

Thanks.

**Comment by tstuefe** [comment]
> This seems okay though may be of limited use if the crash is from a signal as IIRC that in itself will result in various signals getting blocked whilst we remain in the signal-handler context.

True, but its interesting to know if we block what we think we block :-)

**Comment by jdksjolen** [comment]
This LGTM. It fails tier1 tests because it doesn't include the commit that problem lists runtime/os/TestTracePageSizes.java on linux-x64. Merging to master should fix that issue.

**Comment by tstuefe** [comment]
> 



> This LGTM. It fails tier1 tests because it doesn't include the commit that problem lists runtime/os/TestTracePageSizes.java on linux-x64. Merging to master should fix that issue.

Thanks @jdksjolen ! Mind hitting the approve button?

**Review by jdksjolen — APPROVED** [review]

**Pull request closed without merging**